lib/python/ray/worker.py

```diff
@@ -836,20 +836,24 @@ def func_executor(arguments):
       start_time = time.time()
       result = func(*arguments)
       end_time = time.time()
-      check_return_values(func_call, result) # throws an exception if result is invalid
+      check_return_values(func_invoker, result) # throws an exception if result is invalid
       _logger().info("Finished executing function {}, it took {} seconds".format(func.__name__, end_time - start_time))
       return result
-    func_call.executor = func_executor
-    func_call.arg_types = arg_types
-    func_call.return_types = return_types
-    func_call.is_remote = True
+    def func_invoker(*args, **kwargs):
+      """This is returned by the decorator and used to invoke the function."""
+      raise Exception("Remote functions cannot be called directly. Instead of running '{}()', try '{}.remote()'.".format(func_name, func_name))
+    func_invoker.remote = func_call
```

Comment:
I'm not sure what the semantic difference is, but the proper way to bind a *method* to an object is as follows, so you should do it like this:

	>>> a = lambda x: x * 2
	>>> def remote(self): print self
	>>> import types
	>>> a.remote = types.MethodType(remote, a)
	>>> a.remote
	<bound method ?.remote of <function <lambda> at 0x000000000375F828>>

I'm not sure if this will break pickling, but if it does, let me know so I can see if it's easy to fix it.